### ISSUE TITLE
Tiling sticking to wrong screen when dragging to a new one - multi monitor seup

### ISSUE THREAD

**Issue by Carl-Llewellyn** [domferr/tilingshell#211]
**Describe the bug**
When moving a window from one screen to another in a multi-monitor setup, the tiling sticks to the previous screen. This means I cannot move the window from one screen to another using the mouse.

**To Reproduce**

In a multi-monitor set up drag a window from one screen to another.

**Screenshots**

Here's the layout:

![image](https://github.com/user-attachments/assets/148595ad-4bb7-406a-a97c-65ba20b9b9d9)

An application starts here:

![image](https://github.com/user-attachments/assets/a5899214-4b77-46c8-9f0f-4e2115e7ce20)

Dragging it past the screen:

![image](https://github.com/user-attachments/assets/155035b3-0b27-4c40-af2b-d317b8f40956)

It tiles to the original screen instead of just going to where it's left:

![image](https://github.com/user-attachments/assets/c227fefe-506d-45ff-8f10-f69355912a31)

Doesn't happen going down though:

![image](https://github.com/user-attachments/assets/6fd51646-b49d-4625-96f6-a82c59504907)

**Information (please complete the following):**
 - Tiling Shell version: v15.0
 - GNOME version:46.0

**Additional context**
I am using proprietary  NVIDIA RTX A3000 12GB Laptop drivers (550.120) on x11.

**Comment by domferr** [comment]
Hey @Carl-Llewellyn thank you for sharing this. Can you tell me if disabling and enabling the extension fixes the problem?

> Doesn't happen going down though

Sorry, can you elaborate a little bit? Because I didn't understand. Thank you!

**Comment by domferr** [comment]
Hey, I'm looking into this, however currently I'm not in my apartment and I haven't a secondary monitor atm. I'm then guessing a solution. Let me know if it solves the problem!

- GNOME 42, 43 or 44: [GNOME.42-44.tilingshell@ferrarodomenico.com.zip](https://github.com/user-attachments/files/18028917/GNOME.42-44.tilingshell%40ferrarodomenico.com.zip)
- GNOME 45, 46 or 47: [tilingshell@ferrarodomenico.com.zip](https://github.com/user-attachments/files/18028919/tilingshell%40ferrarodomenico.com.zip)

**Comment by domferr** [comment]
Hi, can you check if this is fixed with latest update of Tiling Shell (v15.1)? Thank you

**Comment by achengli** [comment]
Here I have the same issue, I have two monitors, one is 16:9 "4k 27 inches" monitor (3840x2160 resolution) and the primary one is ultrawide 34 inches (3440x1440 resolution @ 21:9 aspect ratio)

When I drag a window to the secondary monitor, tiling shell automatically suggest tiling area in the primary monitor.

**Comment by Carl-Llewellyn** [comment]
> Here I have the same issue, I have two monitors, one is 16:9 "4k 27 inches" monitor (3840x2160 resolution) and the primary one is ultrawide 34 inches (3440x1440 resolution @ 21:9 aspect ratio)
> 
> When I drag a window to the secondary monitor, tiling shell automatically suggest tiling area in the primary monitor.

What version are you on?

**Comment by Carl-Llewellyn** [comment]
> Hi, can you check if this is fixed with latest update of Tiling Shell (v15.1)? Thank you

I'll update and get back

**Comment by Carl-Llewellyn** [comment]
> Hey @Carl-Llewellyn thank you for sharing this. Can you tell me if disabling and enabling the extension fixes the problem?
> 
> > Doesn't happen going down though
> 
> Sorry, can you elaborate a little bit? Because I didn't understand. Thank you!

Yeah disabling and enabling makes it work again. Although it happens to often to make this a viable workaround.

What I meant was that pulling the window to a screen below it didn't seem to cause the issue.

**Comment by achengli** [comment]
> > Here I have the same issue, I have two monitors, one is 16:9 "4k 27 inches" monitor (3840x2160 resolution) and the primary one is ultrawide 34 inches (3440x1440 resolution @ 21:9 aspect ratio)
> > When I drag a window to the secondary monitor, tiling shell automatically suggest tiling area in the primary monitor.
> 
> What version are you on?

I'm using gnome 47 over Fedora 41

**Comment by Carl-Llewellyn** [comment]
> > > Here I have the same issue, I have two monitors, one is 16:9 "4k 27 inches" monitor (3840x2160 resolution) and the primary one is ultrawide 34 inches (3440x1440 resolution @ 21:9 aspect ratio)
> > > When I drag a window to the secondary monitor, tiling shell automatically suggest tiling area in the primary monitor.
> > 
> > 
> > What version are you on?
> 
> I'm using gnome 47 over Fedora 41

I mean version of Tiling Manager

**Comment by achengli** [comment]
> > > > Here I have the same issue, I have two monitors, one is 16:9 "4k 27 inches" monitor (3840x2160 resolution) and the primary one is ultrawide 34 inches (3440x1440 resolution @ 21:9 aspect ratio)
> > > > When I drag a window to the secondary monitor, tiling shell automatically suggest tiling area in the primary monitor.
> > > 
> > > 
> > > What version are you on?
> > 
> > 
> > I'm using gnome 47 over Fedora 41
> 
> I mean version of Tiling Shell

Sorry, I'm using the 15.1 version

**Comment by achengli** [comment]
I tested it now with the current version and it's working well with two monitors

Thank you for the support!
